analytics-integrations/kahuna/src/main/java/com/segment/analytics/internal/integrations/KahunaEcommerceHelper.java

```diff
@@ -0,0 +1,115 @@
+package com.segment.analytics.internal.integrations;
+
+import android.text.TextUtils;
+
+import com.kahuna.sdk.KahunaAnalytics;
+import com.segment.analytics.internal.model.payloads.TrackPayload;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+class KahunaEcommerceHelper {
+
+  // Segment constants
+  private static final String SEG_VIEWED_PRODUCT_CATEGORY = "Viewed Product Category";
+  private static final String SEG_VIEWED_PRODUCT = "Viewed Product";
+  private static final String SEG_ADDED_PRODUCT = "Added Product";
+  private static final String SEG_COMPLETED_ORDER = "Completed Order";
+  private static final String SEG_DISCOUNT_KEY = "discount";
+
+  // Kahuna constants
+  private static final String CATEGORIES_VIEWED = "Categories Viewed";
+  private static final String LAST_VIEWED_CATEGORY = "Last Viewed Category";
+  private static final String LAST_PRODUCT_VIEWED_NAME = "Last Product Viewed Name";
+  private static final String LAST_PRODUCT_VIEWED_ID = "Last Product Viewed Id";
+  private static final String LAST_PRODUCT_ADDED_TO_CART_NAME = "Last Product Added To Cart Name";
+  private static final String LAST_PRODUCT_ADDED_TO_CART_CATEGORY
+          = "Last Product Added To Cart Category";
+  private static final String LAST_PURCHASE_DISCOUNT = "Last Purchase Discount";
+  private static final int MAX_CATEGORIES_VIEWED_ENTRIES = 50;
```

Comment:
I might be wrong here though since I'm not too familiar with how Kahuna works under the hood.